torch/distributed/checkpoint/filesystem.py

```diff
@@ -270,6 +276,16 @@ def _write_files_from_queue(
                     planner.resolve_data,
                     inflight_threshhold=inflight_threshhold,
                 )
+            elif (
+                thread_count == 1
+                and custom_device_mod
+                and custom_device_mod.is_available()
```

Comment:
instead of duping this entire block, work this into the block above `(torch.cuda.is_available() or (custom_device...))`

Reply:
I've fixed it，please check it again